### DISCUSSION TITLE
Configuring IIS for multi-tenant mode

### DISCUSSION THREAD

**Discussion by kudryavtsevda** [OrchardCMS/OrchardCore#11513] [Q&A]
I have an issue regarding to have a lot of absolutely separate sites hosted by orchard. IIS hosts orchard web app. It should contain a lot of small web-sites with their own domain name (site1.mydomain.com, site2.mydomain.com and siteN.mydomain.com).
Am I right that I cannot create separate apppool and web-site in IIS because it is under one orchard process?
So in that case:

1. how to automate creating of tenant with specific domain name (siteN.mydomain.com) for IIS? should I use PS scripts somehow which create new tenant based on recipe?
2. how to have several servers with several IIS and orchard process for load balancing? I see the problem that APP_DATA is one for all orchard process and I have no clue how to makes orchard on "Server2" use APP_Data on "Server1"
3. how to make orchard to use Server1 only for specific set of tenants and Server2 for the rest of tenants? I want to have a very flexible server setup with IIS as a host

**Comment by hishamco** [comment]
You can simulate that in your local environment by editing the `hosts` file which located in **C:\Windows\System32\drivers\etc** you can add virtual domains that point to the same local IP

**Reply by kudryavtsevda** [reply]
yes, I did it but I am talking about automatic approach on windows server with IIS which hosts orchard app

**Reply by hishamco** [reply]
For the first question I think you need execute a PS script or .NET code that creates the entry when the tenant is creating, but this could be used on the development environment

**Comment by ns8482e** [comment]
Use catch all host in IIS by not specifying specific host
Eg just specify mydomain.com to catch site1.mydomain.com and siten.mydomain.com

And update your hosts file (DNS) as @hishamco suggests

**Reply by hishamco** [reply]
@ns8482e seems he asked for an automated process that create everything when the tenant created

**Comment by kudryavtsevda** [comment]
let me be more precise.
I have default site (mysite.com:6001) and second site (second.mysite.com:6001)
I decided to have orchard app on my local machine under mysite.com domain with port 6001
![image](https://user-images.githubusercontent.com/4447809/162898854-304746bc-88af-4bda-bd17-83832c657445.png)

I have `127.0.0.1 mysite.com` mapping in my local hosts file
This is my bindings in IIS 
![image](https://user-images.githubusercontent.com/4447809/162899075-8f006983-fbfb-4788-9836-06de7344ebc0.png)

But second.mysite.com is not available 
![image](https://user-images.githubusercontent.com/4447809/162899334-36a2223a-d7f0-49ab-8915-465dc8494d09.png)

if I put second.mysite.com into hosts file it will work.
But I don't want to add each time {site}.mysite.com into hosts file.
How to configure it properly to be able to add *.mysite.com tenants?

**Reply by ns8482e** [reply]
Your IIS settings seems correct 

this is DNS resolution, host file let’s your browser know where your site hosted and send needed host hearers while connecting.  So you need to have entry for each 

May be you could achieve forwarding all sub domains by having local DNS service
